packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -490,196 +513,185 @@ export function getValidatorApi({
-          consensusBlockValueBuilder: `${consensusBlockValueBuilder}`,
-          blockValueEngine: `${blockValueEngine}`,
-          blockValueBuilder: `${blockValueBuilder}`,
-          shouldOverrideBuilder,
+      if (builder.status === "fulfilled" && engine.status !== "fulfilled") {
+        logger.info("Selected builder block: no engine block produced", {
           slot,
+          durationMs: builder.durationMs,
+          ...getBlockValueLogInfo(builder.value),
         });
-      } else if (fullBlock && !blindedBlock) {
-        executionPayloadSource = ProducedBlockSource.engine;
+
+        return {...builder.value, executionPayloadBlinded: true, executionPayloadSource: ProducedBlockSource.builder};
+      }
+
+      if (engine.status === "fulfilled" && builder.status !== "fulfilled") {
         logger.info("Selected engine block: no builder block produced", {
-          // winston logger doesn't like bigint
-          enginePayloadValue: `${enginePayloadValue}`,
-          consensusBlockValueEngine: `${consensusBlockValueEngine}`,
-          blockValueEngine: `${blockValueEngine}`,
-          shouldOverrideBuilder,
           slot,
+          durationMs: engine.durationMs,
+          ...getBlockValueLogInfo(engine.value),
         });
-      } else if (blindedBlock && !fullBlock) {
-        executionPayloadSource = ProducedBlockSource.builder;
-        logger.info("Selected builder block: no engine block produced", {
-          // winston logger doesn't like bigint
-          builderPayloadValue: `${builderPayloadValue}`,
-          consensusBlockValueBuilder: `${consensusBlockValueBuilder}`,
-          blockValueBuilder: `${blockValueBuilder}`,
-          shouldOverrideBuilder,
+
+        return {...engine.value, executionPayloadBlinded: false, executionPayloadSource: ProducedBlockSource.engine};
+      }
+
+      if (engine.status === "fulfilled" && builder.status === "fulfilled") {
+        const engineBlockValue = engine.value.executionPayloadValue ?? 0n + engine.value.consensusBlockValue ?? 0n;
+        const builderBlockValue = builder.value.executionPayloadValue ?? 0n + builder.value.consensusBlockValue ?? 0n;
+        const {
+          executionPayloadValue: engineExecutionPayloadValue,
+          consensusBlockValue: engineConsensusBlockValue,
+          blockTotalValue: engineBlockTotalValue,
+        } = getBlockValueLogInfo(engine.value);
+        const {
+          executionPayloadValue: builderExecutionPayloadValue,
+          consensusBlockValue: builderConsensusBlockValue,
+          blockTotalValue: builderBlockTotalValue,
+        } = getBlockValueLogInfo(builder.value);
+
+        const executionPayloadSource = selectBlockProductionSource({
+          builderBlockValue,
+          engineBlockValue,
+          builderBoostFactor,
+          builderSelection,
+        });
+
+        logger.info(`Selected executionPayloadSource=${executionPayloadSource} block`, {
```

Comment:
These are logged as info, we might consider a bit more user friendly format, e.g. do we need to log block values in wei? The wei values are also quite hard to compare as those are just huge numbers.

Could apply same formatting as in publish block log

https://github.com/ChainSafe/lodestar/blob/01d47b9a0901c9f03a01ef68c819f4e592cbfd8b/packages/validator/src/services/block.ts#L223